### ISSUE TITLE
[bazel] mask rom verilator test hangs on uart output

### ISSUE THREAD

**Issue by drewmacrae** [lowRISC/opentitan#9784]
I'm not sure how to get a call stack, but if I look at the trace_core_00000000.log and a disassembly file, I see
mask_rom_main
called
shutdown_finalize
called
printf
called 
uart_putchar
which never returned.

I can't tell if my console isn't working properly or if putchar (which appears to be polling for idle is examining a module with a stopped clock.)

**Comment by drewmacrae** [comment]
Looking at shutdown_finalize I think it's trying to report the error we expect here https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/lib/shutdown.c;drc=319dbc899c8fcab7bcf516ed756f291146696fd0;l=272

**Comment by drewmacrae** [comment]
This failure appears as a test timing out 
`//sw/device/silicon_creator/mask_rom:verilator_mask_rom_test            TIMEOUT in 300`

**Comment by drewmacrae** [comment]
I'm hitting this in Verilator 4.210

**Comment by drewmacrae** [comment]
I don't see the UART transmitting or going idle when I look at a trace. The FIFO has a single byte but cio_tx_en_o appears to remain low through the test. I wonder if we need to set that somehow.

**Comment by drewmacrae** [comment]
Oooh, the dif_uart appears to set CTRL.LLPBK but the uart driver this test uses doesn't I bet that's the problem.

**Comment by drewmacrae** [comment]
The NCO isn't set, so the baudrate generator is inactive.

**Comment by drewmacrae** [comment]
It looks like the driver expects that the uart get initialized before it's used, and tries to use it to report an unsigned flash image at the end of the mask rom test. 



I don't see discussion of initializing the uart in the mask_rom.

**Comment by msfschaffner** [comment]
Is this still failing? If so, do other folks have to be pulled in to help debugging?

**Comment by drewmacrae** [comment]
I think @timothytrippel did some work on this. I'm not familiar with the POR for Mask ROM debuggability WRT printing details on test failures.

**Comment by drewmacrae** [comment]
Open Questions: When a test in Mask ROM fails how should it report the failure? (Especially if the UART isn't yet initialized)

**Comment by timothytrippel** [comment]
@cfrantz fixed the UART timeout issue in #11300.

However, now that we removed support for RSA with exponent 3 (#10141), the sigverify takes much longer in Verilator. So long in fact that the 1 hour time limit of an `opentitan_functest` causes a test timeout.

At one point I had it booting and running a mask ROM E2E test, but I think this was before the exponent 3 test key was removed. I need to verify how long it takes to run. But I think that is an orthogonal issue, and this issue can be closed? wdyt?

To answer your question about when a mask ROM test fails, it depends, but usually there is a boot fault message of sorts. See @milesdai 's recent PR #11620 with some mask ROM tests that includes the regex we pass to opentitantool console command to define what a test success vs failure message can look like.
